ansible/library/get_ip_in_range.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/python
+from netaddr import *
+import sys
+import ipaddress
+
+DOCUMENTATION = '''
+module:         get_ip_in_range
+version_added:  "1.0"
+short_description: Get certain number of ips within a prefix
+description:
+    - Generate certain number of unique IP withn a subnet
+    - Generated ips will be inserted into the 'generated_ips' key
+options:
+   num:
+      description:
+          - set to the number of ip that needs to be generated.
+      required: true
+   prefix:
+      description:
+          - the required ip range in prefix format.
+      required: true
+   exclude_ips:
+      description:
+          - the ips within the prefixs that are excluded.
+          - the format is ips separated by whitespace eg: 10.0.0.0 10.0.0.1
+      required: false
+'''
+
+EXAMPLES = '''
+- name: Get IP in range
+  get_ip_in_range: num={{num}} prefix={{prefix}}
+'''
+
+class IpRangeModule(object):
+    def __init__(self):
+        self.module = AnsibleModule(
+            argument_spec=dict(
+              num=dict(required=True, type='int'),
+              prefix=dict(required=True),
+              exclude_ips=dict(required=False),
+            ),
+            supports_check_mode=True)
+
+        self.out = None
+        self.facts = {}
+
+        return
+
+    def run(self):
+        """
+            Main method of the class
+        
+        """
+        m_args = self.module.params
+        exclude_ips = []
+        if m_args['exclude_ips'] is not None:
+            ip_list = m_args['exclude_ips'].split() 
```

Comment:
You can define this parameter as a list. Something like this:
exclude_ips=dict(required=False, default=[], type='list'),